src/test/fuzz/compression_roundtrip.cpp

```diff
@@ -0,0 +1,482 @@
+	std::vector<secp256k1_keypair> keypairs;
+	CMutableTransaction mtx;
+
+	mtx.nVersion = fdp.ConsumeIntegral<uint32_t>();
+	mtx.nLockTime = fdp.ConsumeIntegral<uint32_t>();
+
+	uint32_t total = 0;
+	std::vector<CScript> input_scripts;
+	std::vector<uint256> txids;
+	std::vector<CCompressedTxId> compressed_txids;
+	std::vector<int> used_indexs;
+
+
+	// GENERATE INPUTS //
+	LIMITED_WHILE(total == 0 || fdp.ConsumeBool(), static_cast<unsigned char>(unspent_transactions.size()-1)) {
+		int index = fdp.ConsumeIntegralInRange<int>(0, unspent_transactions.size()-1);
+		if (std::find(used_indexs.begin(), used_indexs.end(), index) != used_indexs.end())
+			break;
+		used_indexs.push_back(index);
+		uint256 txid = std::get<0>(unspent_transactions.at(index));
+		keypairs.push_back(std::get<1>(unspent_transactions.at(index)));
+		uint32_t vout = std::get<2>(unspent_transactions.at(index));
+		input_scripts.push_back(std::get<4>(unspent_transactions.at(index)));
+		txids.push_back(std::get<5>(unspent_transactions.at(index)));
+		compressed_txids.push_back(std::get<6>(unspent_transactions.at(index)));
+		
+		total += std::get<3>(unspent_transactions.at(index));
+
+		CTxIn in;
+		in.prevout = COutPoint{txid, vout};
+		in.nSequence = fdp.ConsumeIntegral<uint32_t>();
+		mtx.vin.push_back(in);
+	}
+
+
+	// GENERATE OUTPUTS //
+	uint32_t remaining_amount = total;
+	bool sign = true;
+	LIMITED_WHILE(remaining_amount > 2000 && (fdp.ConsumeBool() || mtx.vout.size() == 0), 10000) {
+		CTxOut out;	
+		uint32_t limit = pow(2, 16);
+		int range_amount;
+		if (remaining_amount > limit) {
+			range_amount = limit-1000;
+		} else {
+			range_amount = remaining_amount-1000;
+		}
+		uint16_t amount = fdp.ConsumeIntegralInRange<uint16_t>(1, range_amount);
+		remaining_amount -= amount;
+		out.nValue = amount;
+
+		std::vector<unsigned char> secret_key = fdp.ConsumeBytes<uint8_t>(32);
+		if (secret_key.size() != 32) return;
+		secp256k1_keypair out_kp;
+		if (!secp256k1_keypair_create(ctx, &out_kp, &secret_key[0])) return;
+
+		CScript out_scriptPubKey;
+
+		int script_type = fdp.ConsumeIntegralInRange<uint16_t>(0, 7);
+		if (script_type == 7) 
+			sign = false;
+		std::vector<unsigned char> custom_script_bytes = fdp.ConsumeBytes<uint8_t>(128);
+		out.scriptPubKey = rpc->GenerateDestination(ctx, out_kp, script_type, custom_script_bytes);
+		mtx.vout.push_back(out);
+	}
+	if (mtx.vout.size() == 0) return;
+	
+	if (sign)
+		assert(rpc->SignTransaction(ctx, mtx, keypairs));
+
+	std::cout << "sttx: " << CTransaction(mtx).ToString() << std::endl;
+
+    const CTransaction tx = CTransaction(mtx);
+    CCompressedTransaction compressed_transaction = CCompressedTransaction(ctx, tx, compressed_txids, input_scripts);
+	
+    std::cout << "ctx: " << compressed_transaction.ToString() << std::endl;
+
+    CDataStream stream(SER_DISK, 0);
+    compressed_transaction.Serialize(stream);
+    CCompressedTransaction uct = CCompressedTransaction(deserialize, stream);
+    
```

Comment:
In 1d376496fe9ac7110d0fac773ae5b5d9ef32aa59:

Trailing whitespace. You can find this by running `git grep ' $'` on every commit and deleting it where you see it.

Reply:
Fixed in all commits, Is there a way to git grep for just my changes?

Reply:
Sadly no, but these really shouldn't occur anywhere...I'm surprised to see so many in git master.

You can at least restrict to the `src/` and `test/` directories.